unikernel.ml

```diff
@@ -973,62 +1009,254 @@ struct
+                ~content:
+                  (Error_page.error_layout
+                     {
+                       code = 500;
+                       success = false;
+                       title = "Albatross Error";
+                       data =
+                         `String
+                           ("An error occured trying to fetch " ^ name
+                          ^ "from albatross: " ^ err);
+                     })
+                ~icon:"/images/robur.png" ())
+             `Internal_server_error)
+    | Some unikernel, None -> (
+        (Albatross.query_console ~domain:user.name albatross ~name >|= function
+         | Error err ->
+             Logs.warn (fun m ->
+                 m "error querying console of albatross: %s" err);
+             []
+         | Ok (_, console_output) -> console_output)
+        >>= fun console_output ->
+        let now = Ptime.v (P.now_d_ps ()) in
+        generate_csrf_token store user now reqd >>= function
+        | Ok csrf ->
+            Lwt.return
+              (reply reqd ~content_type:"text/html"
+                 (Dashboard.dashboard_layout ~csrf user
+                    ~content:
+                      (Unikernel_single.unikernel_single_layout
+                         ~unikernel_name:name unikernel now console_output)
+                    ~icon:"/images/robur.png" ())
+                 ~header_list:[ ("X-MOLLY-CSRF", csrf) ]
+                 `OK)
+        | Error err ->
+            Lwt.return
+              (reply reqd ~content_type:"text/html"
+                 (Guest_layout.guest_layout ~page_title:"500 | Mollymawk"
+                    ~content:(Error_page.error_layout err)
+                    ~icon:"/images/robur.png" ())
+                 `Internal_server_error))
+    | _ ->
+        Logs.err (fun m ->
+            m "n unknown error occured while fetching %s from albatross." name);
+        Middleware.redirect_to_error
+          ~data:
+            (`String
+              "An unknown error occured while fetching this unikernel from \
+               albatross. ")
+          ~title:(name ^ " update Error") ~api_meth:false `Internal_server_error
+          reqd ()
+
+  let unikernel_prepare_update albatross store name reqd http_client
+      ~json_dict:_ (user : User_model.user) =
+    (* TODO use uuid in the future *)
+    user_unikernel albatross ~user_name:user.name ~unikernel_name:name
+    >>= fun unikernel_info ->
+    match unikernel_info with
+    | None, Some err ->
+        Middleware.redirect_to_error
+          ~data:
+            (`String
+              ("An error occured while fetching " ^ name
+             ^ " from albatross with error " ^ err))
+          ~title:"Albatross Error" ~api_meth:false `Internal_server_error reqd
+          ()
+    | Some (unikernel_name, unikernel), None -> (
+        Builder_web.send_request http_client
+          (Builder_web.base_url ^ "/hash?sha256=" ^ Ohex.encode unikernel.digest)
```

Comment:
as suggested in `Builder_web`, we can drop the `Builer_web.base_url` here and always prefix in `send_request`.